dao/patient/etl-patient-dao.js

```diff
@@ -1,309 +1,354 @@
 
-    },
-    getClinicalNotes: function getClinicalNotes(request, callback) {
-      var patientEncounters=noteService.getPatientEncounters(request);
-      var patientHivSummary=noteService.getHivSummary(request);
-      var patientVitals=noteService.getVitals(request);
+    var queryParts = {
+      columns: request.query.fields || "t1.*, t2.cur_arv_meds",
+      table: "etl.flat_labs_and_imaging",
+      joins: [
+        ['etl.flat_hiv_summary', 't2', 't1.encounter_id = t2.encounter_id']
+      ],
+      where: ["t1.uuid = ?", uuid],
+      order: order || [{
+        column: 'test_datetime',
+        asc: false
+      }],
+      offset: request.query.startIndex,
+      limit: request.query.limit
+    };
 
-      Promise.all([patientEncounters,patientHivSummary,patientVitals]).then(function(data) {
-          var encounters = data[0].results;
-          var hivSummaries = data[1].result;
-          var vitals = data[2].result;
-          noteService.generate(encounters,hivSummaries,vitals,function (notes) {
-              callback({notes:notes,status:'notes generated'});
-          });
+    db.queryServer_test(queryParts, function(result) {
 
-      })
-      .catch (function(e) {
-          // Return empty json on error
-          console.log('Error',e);
-          callback({notes:[],status:'error generating notes'});
+      _.each(result.result, function(row) {
+        row.tests_ordered = helpers.getTestsOrderedNames(row.tests_ordered);
+        row.cur_arv_meds = helpers.getARVNames(row.cur_arv_meds);
       });
-    },
-    getPatientData: function getPatientData(request, callback) {
-      var uuid = request.params.uuid;
-      var order =  helpers.getSortOrder(request.query.order);
+      callback(result);
+    });
+  }
+  
+  function getPatient(request, callback) {
+    var uuid = request.params.uuid;
+    var order =  helpers.getSortOrder(request.query.order);
 
-      var queryParts = {
-        columns: request.query.fields || "t1.*, t2.cur_arv_meds",
-        table: "etl.flat_labs_and_imaging",
-        joins: [
-          ['etl.flat_hiv_summary', 't2', 't1.encounter_id = t2.encounter_id']
-        ],
-        where: ["t1.uuid = ?", uuid],
-        order: order || [{
-          column: 'test_datetime',
-          asc: false
-        }],
-        offset: request.query.startIndex,
-        limit: request.query.limit
-      };
+    var queryParts = {
+      columns: request.query.fields || "*",
+      table: "etl.flat_hiv_summary",
+      where: ["uuid = ?", uuid],
+      order: order || [{
+        column: 'encounter_datetime',
+        asc: false
+      }],
+      offset: request.query.startIndex,
+      limit: request.query.limit
+    };
 
-      db.queryServer_test(queryParts, function(result) {
+    db.queryServer_test(queryParts, function(result) {
+      callback(result);
+    });
+  }
+  
+  function getPatientStgetPatientCountGroupedByLocationatics(request, callback) {
```

Comment:
This is what I was asking Manu about. The name surely does not make sense whatsoever. May be it was one of those mouse moves that screw things up